### PR TITLE
[MAINTENANCE] Update `ubuntu-latest` to specific version in our Github Actions CI/CD

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What is the issue? 
* Occasionally PRs will hang in a "Waiting for status to be reported" state, for Github actions, even after approval and auto-merge enabled.  An example is #8499, which was stuck in that state for 3+ hours on 8/3/23.
 
Screenshot 2023-08-04 at 4 41 02 PM
* This is an issue faced by others [(Example 1)](https://github.com/orgs/community/discussions/56360), [(Example 2)](https://github.com/orgs/community/discussions/58508) and [(Example 3)](https://github.com/orgs/community/discussions/26693), with the issue either being an incident with github, or using `ubuntu-latest` rather than a specific version of ubuntu. 

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,14 +19,14 @@ on:
 
 jobs:
   ci-does-not-run-on-draft-pull-requests:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```
Comment (Kilo59) [inline .github/workflows/ci.yml:22]: Why are we making this change?

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml:22]: updating description now

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml:22]: Great question. It seems to be the cause of why some PRs get stuck in the "waiting for status to be reported" state (which I've been seeing with #8499). 

Other users have been reporting similar problems (which I've linked in the PR description), and for us at least, it appears that the `ubuntu-23.04` image is causing the problem for us.

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml:22]: trying again with `23.04` to make sure, and it hangs. Appears to work with `22.04` only

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml:22]: Yup, it seems to work with `22.04` only

**Review by Kilo59 — CHANGES_REQUESTED** [review]
We should discuss this more before merging this. I think some of the fixes provided by this may be coincidences. 
Users in those threads also report the opposite behavior (moving from a pinned Ubuntu version being fixed by moving to `ubuntu-latest`).

Also, it could be an intermittent issue that will be resolved upstream.

**Comment by Shinnnyshinshin** [comment]
closing, as we keep track of how often this occurs, and whether this is the correct approach. cc @Kilo59 @billdirks

**Pull request closed without merging**